pkg/ccl/logictestccl/tests/local/local_test: TestCCLLogic_as_of failed

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/ef3610bb-6759-4c43-ab32-73a0c39f4ce4?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [d9fa871053bb1db45308ff969d3704555adfed9f](https://github.com/cockroachdb/cockroach/commits/d9fa871053bb1db45308ff969d3704555adfed9f):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of2731288075
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:2695: 
         
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:95: 
        expected:
        pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[11:57:11] --- done: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 26 tests, 1 failures
    logic.go:4225: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: error while processing
    logic.go:4225: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of2731288075
--- FAIL: TestCCLLogic_as_of (4.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44924

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/79c893aa-5ebc-4c4c-aa17-3b0e4cc0d55c?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [b77653f98b4ce56425f7b0997671d8414051ea54](https://github.com/cockroachdb/cockroach/commits/b77653f98b4ce56425f7b0997671d8414051ea54):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of1743848246
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:2695: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:95: 
        expected:
        pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[12:25:30] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 26 tests, 1 failures
    logic.go:4225: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: error while processing
    logic.go:4225: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of1743848246
--- FAIL: TestCCLLogic_as_of (7.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/91b9bbe9-e44f-4a24-b21e-7a37f30041ce?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [f2ce52ff7cc3b4680e0fec2ef844baad49042d6a](https://github.com/cockroachdb/cockroach/commits/f2ce52ff7cc3b4680e0fec2ef844baad49042d6a):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of408212826
    test_log_scope.go:76: use -show-logs to present logs inline
[12:20:07] setting distsql_workmem='94638B';
    logic.go:2695: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:95: 
        expected:
        pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[12:20:09] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 26 tests, 1 failures
    logic.go:4225: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: error while processing
    logic.go:4225: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of408212826
--- FAIL: TestCCLLogic_as_of (4.85s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/d295ec38-e4a6-4b7c-bdc1-a09ee1be59f9?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [749b563a7b5957b2db487c6e8e2fdeed8630103d](https://github.com/cockroachdb/cockroach/commits/749b563a7b5957b2db487c6e8e2fdeed8630103d):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of913053517
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:2695: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:95: 
        expected:
        pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[13:03:27] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 26 tests, 1 failures
    logic.go:4225: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: error while processing
    logic.go:4225: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:98: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of913053517
--- FAIL: TestCCLLogic_as_of (5.64s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/as_of
+++ b/pkg/ccl/logictestccl/testdata/logic_test/as_of
@@ -48,7 +48,7 @@ SELECT * FROM t
 
 # LSC and DSC would return slightly different error message when attempting to create a
 # database as of system time follower_read_timestamp() soon after a node has started.
-statement error pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist)
+statement error pq: (cannot execute CREATE DATABASE in a read-only transaction|referenced descriptor ID 1: looking up ID 1: descriptor not found|database \"\[1\]\" does not exist|role/user \"root\" does not exist)
 CREATE DATABASE IF NOT EXISTS d2
 
 statement error pgcode 3D000 pq: database "test" does not exist

